posthog/tasks/detect_alerts_anomalies.py

```diff
@@ -0,0 +1,77 @@
+from typing import cast
+
+import structlog
+from celery import shared_task
+from django.utils import timezone
+
+from posthog.email import EmailMessage
+from posthog.hogql_queries.legacy_compatibility.filter_to_query import filter_to_query
+from posthog.hogql_queries.query_runner import get_query_runner
+from posthog.models import Alert, AnomalyCondition
+from posthog.schema import HogQLQueryResponse
+
+logger = structlog.get_logger(__name__)
+
+
+def check_all_alerts() -> None:
+    alerts = Alert.objects.all().only("id")
+    for alert in alerts:
+        logger.info("scheduling alert", alert_id=alert.id)
+        check_alert.delay(alert.id)
+
+
+@shared_task(ignore_result=True)
+def check_alert(id: int) -> None:
+    alert = Alert.objects.get(pk=id)
```

Comment:
Please move task bodies into separate functions for testability and readability.

Reply:
> for testability

I wouldn't say it hurts testability - one can call a task as a simple function

> and readability

I could argue that it doesn't help readability to have a function that just calls another function. Also, since we will have a separate task for sending notifications, we will have check_alert_task + check_alert + send_notification_task + send_notification. IMO it becomes cumbersome, I'll do it, but I'll be glad to revert it if you agree with me.